src/leetCodeSolutionProvider.ts

```diff
@@ -0,0 +1,124 @@
+                "</code></pre>",
+            ].join("\n");
+        };
+    }
+
+    public async show(solutionString: string, problem: IProblem): Promise<void> {
+        if (!this.panel) {
+            this.panel = window.createWebviewPanel("leetCode", "Top voted solution", ViewColumn.Active, {
+                retainContextWhenHidden: true,
+                enableFindWidget: true,
+                localResourceRoots: [vscode.Uri.file(path.join(this.markdownPath, "media"))],
+            });
+
+            this.panel.onDidDispose(() => {
+                this.panel = undefined;
+            }, null, this.context.subscriptions);
+        }
+
+        this.solution = this.parseSolution(solutionString);
+        this.panel.title = problem.name;
+        this.panel.webview.html = this.getWebViewContent(this.solution);
+        this.panel.reveal(ViewColumn.Active);
+    }
+
+    public dispose(): void {
+        if (this.panel) {
+            this.panel.dispose();
+        }
+    }
+
+    private parseSolution(raw: string): Solution {
+        const solution: Solution = new Solution();
+        // [^] matches everything including \n, yet can be replaced by . in ES2018's `m` flag
+        raw = raw.slice(1); // skip first empty line
+        [solution.title, raw] = raw.split(/\n\n([^]+)/); // parse title and skip one line
+        [solution.url, raw] = raw.split(/\n\n([^]+)/); // parse url and skip one line
+        [solution.lang, raw] = raw.match(/\* Lang:\s+(.+)\n([^]+)/)!.slice(1);
+        [solution.author, raw] = raw.match(/\* Author:\s+(.+)\n([^]+)/)!.slice(1);
+        [solution.votes, raw] = raw.match(/\* Votes:\s+(\d+)\n\n([^]+)/)!.slice(1);
+        solution.body = raw;
+        return solution;
+    }
+
+    private codeHighlighter(code: string, lang: string | undefined): string {
+        if (!lang) {
+            lang = this.solution.lang;
+        }
+        if (hljs.getLanguage(lang)) {
+            try {
+                return hljs.highlight(lang, code, true).value;
+            } catch (error) { /* do not highlight */ }
+        }
+        return ""; // use external default escaping
+    }
+
+    private getMarkdownStyles(): vscode.Uri[] {
+        const stylePaths: string[] = require(path.join(this.markdownPath, "package.json"))["contributes"]["markdown.previewStyles"];
+        return stylePaths.map((p: string) => vscode.Uri.file(path.join(this.markdownPath, p)).with({ scheme: "vscode-resource" }));
+    }
+
+    private getWebViewContent(solution: Solution): string {
+        const styles: string = this.getMarkdownStyles()
+            .map((style: vscode.Uri) => `<link rel="stylesheet" type="text/css" href="${style.toString()}">`)
+            .join("\n");
+        const { title, url, lang, author, votes } = solution;
+        const head: string = this.markdown.render(`# [${title}](${url})`);
+        const auth: string = `[${author}](https://leetcode.com/${author}/)`;
```

Comment:
So the solutions are always from the English version of LeetCode right?

Reply:
After test, the `solution.discuss` indeed fetch solutions only from leetcode.com. After switching to our own library, we may be able to freely choose endpoint.